File: LeetCode/Medium/0649-dota2-senate/0649-dota2-senate.go
func predictPartyVictory(senate string) string {
    turn := make([]byte, len(senate))
    r := 0
    d := 0
    rSkip := 0
    dSkip := 0
    i := 0
    
    for i := 0; i < len(senate); i++ {
        if senate[i] == 'R' { 
            r++ 
        } else { d++ }
        turn[i] = senate[i]
    }
    
    for r > 0 && d > 0 {
        temp := turn[i]
        i++
        
        if temp == 'R' {
            if rSkip > 0 {
                rSkip--
                continue
            }
            d--
            dSkip++
        } else {
            if dSkip > 0 {
                dSkip--
                continue
            }
            r--
            rSkip++
        }
        
        turn = append(turn, temp)
    }
    
    if r > 0 { return "Radiant" }
    return "Dire"
}

File: LeetCode/Medium/0649-dota2-senate/0649-dota2-senate.py
class Solution:
    def predictPartyVictory(self, senate: str) -> str:
      turn = deque(list(senate))
      r = senate.count("R")
      d = senate.count("D")
      r_skip = 0
      d_skip = 0

      while r and d and turn:
        temp = turn.popleft()
        
        if temp == "R":
          if r_skip:
            r_skip -= 1
            continue
          d -= 1
          d_skip += 1
            
        else:
          if d_skip:
            d_skip -= 1
            continue
          r -= 1
          r_skip += 1
          
        turn.append(temp)

      return "Radiant" if r else "Dire"
      